analyzers/tests/SonarAnalyzer.Test/Rules/AspNet/ControllersHaveTooManyResponsibilitiesTest.cs

```diff
@@ -0,0 +1,51 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis.CSharp;
+using SonarAnalyzer.Rules.CSharp;
+
+namespace SonarAnalyzer.Test.Rules;
+
+[TestClass]
+public class ControllersHaveTooManyResponsibilitiesTest
+{
+    private readonly VerifierBuilder builder =
+        new VerifierBuilder<ControllersHaveTooManyResponsibilities>();
+
+#if NET
+    private static IEnumerable<MetadataReference> AspNetCoreReferences =>
```

Comment:
When you combine your three suggestions "use a static builder", "initialize the references in the builder" and "use a static field for references" you get a potential initialization issue, since the order of initialization of inline static fields is not guaranteed, and `references` may not be initialized while you are initializing `builder`.

You never fall into such traps with static fields initialization if you use a property. The property is anyway only accessed once, so there is virtually no performance loss once the code is JITed.